### DISCUSSION TITLE
Function scoring

### DISCUSSION THREAD

**Discussion by sandstrom** [meilisearch/product#205] [Feedback & Feature Proposal]
ElasticSearch has something called "Function Scoring", which can be used to e.g. gradually phase out certain records from search results. Very useful for e.g. entries that should gradually grow stale.

A few common use-cases:

- Rank nearby restaurants/hotels/etc higher
- Rank old records (or new records) higher
- Reddit-style tradeoff, recency vs. many votes
- Leniency in price-range queries (show some things outside the range, but only when they are close; taper heavily for those far outside)

We use it in ~50% of our ElasticSearch queries. Just wanted to see if this is anything you've planned on introducing?

https://www.elastic.co/es/blog/found-function-scoring#decay-functions-in-elasticsearch

**Comment by gmourier** [comment]
Hello sandstorm 👋

Thank you for this feedback and proposal. 

It would be great to have this to make the selection of results more "intelligent" according to the criteria chosen by the user. Am I right?

Do you use one function more often than another?

Thanks

**Reply by sandstrom** [reply]
I'd say a few common ones are:

- Rank nearby restaurants/hotels/etc higher
- Rank old records (or new records) higher
- Reddit-style tradeoff, recency vs. many votes
- Leniency in price-range queries (show some things outside the range, but only when they are close; taper heavily for those far outside)

*(on the top of my mind, you can find many more if you google "ElasticSearch function scoring use-case")*

**Comment by sandstrom** [comment]
@gmourier just wanted to check in and see if there are any updates on this? 

This is one area that's currently blocking us from migrating off ElasticSearch to Meili.

**Reply by gmourier** [reply]
Hi @sandstrom 👋

Sorry for the late reply. This topic has not progressed. It is not a priority today even if I see a particular interest to have this type of operation in the future.

Thanks!

**Comment by sandstrom** [comment]
@gmourier Congrats on the new website! Looks awesome! 🏅 

Just wanted to check in on this one, any chance you'll add this to your roadmap later this year? (decay and function scoring)

https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-function-score-query.html#function-decay

Now that delete by query and multi-index search has been added, this is the last blocker until we can move off of ElasticSearch.

**Comment by macraig** [comment]
Hello everyone 👋

We just released a 🧪 prototype that allows **boosting results that match a custom filter**, and we'd love your feedback.

### How to use the prototype?
See https://www.notion.so/meilisearch/boostingFilter-API-usage-20aae06bc85e41dba828a90331a69f2c

⚠️ We do NOT recommend using this prototype in production. This is for test purposes only.

Feedback and bug reporting when using this prototype are encouraged! Thanks in advance for your involvement. It means a lot to us ❤️

**Reply by sandstrom** [reply]
Thanks for letting us know, but as the OP I'd say that custom filters is different from function scoring.

**Reply by macraig** [reply]
Hi @sandstrom , thanks for your feedback. The new search parameter, `boostingFilter`, can be used to express boosting filter criteria. Documents matching these criteria are pushed to the top of the search results (*boosted; promoted*) but can also be pushed to the back of the search results using negative criteria (*buried*). Unlike the usual `filter`, boosting filters don't remove documents from search results when the query doesn't match them. The results are divided into two sets: firstly, those that match the boosting filters, and secondly, those that do not.

Even though it's not the same feature, I believe it would help the use cases you described initially. Let me know if that's not the case!

**Reply by sandstrom** [reply]
Thanks, but that's not the case.
